pkg/api/configsync/register.go

```diff
@@ -23,7 +23,8 @@ const (
 	// ConfigSyncPrefix is the prefix for all ConfigSync annotations and labels.
 	ConfigSyncPrefix = GroupName + "/"
 
-	// FieldManager is the field manager name for server-side apply.
+	// FieldManager is the field manager name used by the reconciler.
```

Comment:
We already have code to ensure controllers don't fight with each other, and it will be even less likely with watch label filters. Plus, there's no way to tell config sync that you only want to patch an existing object. If you tried, it would probably create a race condition that would fail due to missing required fields until created by a reconciler with a fully valid spec. Plus, we would definitely want to test that kind of new feature if we added support for it. So I don't think we need to do future-proofing for it now.

Reply:
TLDR: Using Config Sync to patch existing objects could be a future feature addition, but it would require a lot more thought. And we can always change the field managers later without much problem.